core/src/main/java/org/apache/accumulo/core/client/impl/TabletLocatorImpl.java

```diff
@@ -330,8 +348,16 @@ else if (!lookupFailed)
         tabletLocations.add(tl);
       }
 
-      for (TabletLocation tl2 : tabletLocations) {
-        TabletLocatorImpl.addRange(binnedRanges, tl2.tablet_location, tl2.tablet_extent, range);
+      // Ensure the extents found are non overlapping and have no holes. When reading some extents
+      // from the cache and other from the metadata table in the loop above we may end up with
+      // non-contiguous extents. This can happen when a subset of exents are placed in the cache and
+      // then after that merges and splits happen.
+      if (isContiguous(tabletLocations)) {
```

Comment:
I'm wondering if this check should be moved to the end of loop `l1`. Multiple disjoint ranges could be sent into this method and I'm not sure that's taken into account here. I'm assuming that you want to check that you have a contiguous list of TabletLocations for each Range.

Reply:
The code is checking that the extents found for each range are contiguous.  If not, the range is added to the list of failed ranges.  I think the check is in the right place.  Looking back at the code I noticed it was not setting a boolean that other places set when adding a failed range, so I added that.

Reply:
Ok, you are right, I missed a bracket, so I thought that this was outside of loop `l1`.